File: #11_Containe_With_Most_Water/Solution.py
class Solution:
    # @param {integer[]} height
    # @return {integer}
    def maxArea(self, height):
        if not height:
        	return 0
        re = 0
        i, j = 0, len(height) - 1
        while i < j:
        	re = max(re, (j - i) * min(height[i], height[j]))
        	if height[i] <= height[j]:
        		i += 1
        	elif height[i] >= height[j]:
        		j -= 1
        	else:
        		i += 1
        		j -= 1
        return re


sol = Solution()
t = [3, 7, 9, 2]
print sol.maxArea(t)

File: #11_Containe_With_Most_Water/Solution.java
public class Solution {
    public int maxArea(int[] height) {
        if (height == null || height.length == 0) return 0;
        int max = 0;
        int i = 0; int j = height.length - 1;
        while (i < j) {
        	max = Math.max(max, (j - i) * Math.min(height[i], height[j]));
        	if (height[i] < height[j]) i++;
        	else j--;
        }
        return max;
    }

    public static void main(String[] args) {
    	Solution sol = new Solution();
    	int[] t = new int[5];
    }
}